packages/iris-grid/src/sidebar/visibility-ordering-builder/sortable-tree/SortableTree.tsx

```diff
@@ -0,0 +1,350 @@
+  getProjection,
+  getChildCount,
+  removeItem,
+  removeChildrenOf,
+  setProperty,
+} from './utilities';
+import type { FlattenedItem, SensorContext, TreeItem } from './types';
+import { sortableTreeKeyboardCoordinates } from './keyboardCoordinates';
+import { SortableTreeItem } from './SortableTreeItem';
+import type { Props as SortableTreeItemProps } from './SortableTreeItem';
+import styles from './TreeItem.module.scss';
+
+/**
+ * An extended "PointerSensor" that prevent some
+ * interactive html element(button, input, textarea, select, option...) from dragging
+ */
+class PointerWithInteraction extends PointerSensor {
+  static activators = [
+    {
+      eventName: 'onPointerDown' as const,
+      handler: ({ nativeEvent: event }: PointerEvent) => {
+        if (
+          !event.isPrimary ||
+          event.button !== 0 ||
+          isInteractiveElement(event.target as Element)
+        ) {
+          return false;
+        }
+
+        return true;
+      },
+    },
+  ];
+}
+
+class KeyboardWithInteraction extends KeyboardSensor {
+  static activators = [
+    {
+      eventName: 'onPointerDown' as const,
+      handler: ({ nativeEvent: event }: PointerEvent) => {
+        if (
+          !event.isPrimary ||
+          event.button !== 0 ||
+          isInteractiveElement(event.target as Element)
+        ) {
+          return false;
+        }
+
+        return true;
+      },
+    },
+  ];
+}
+
+function isInteractiveElement(element: Element | null) {
+  const interactiveElements = [
+    'button',
+    'input',
+    'textarea',
+    'select',
+    'option',
+  ];
+  if (
+    element?.tagName != null &&
+    interactiveElements.includes(element.tagName.toLowerCase())
+  ) {
+    return true;
+  }
+
+  return false;
+}
+
+const measuring = {
+  droppable: {
+    strategy: MeasuringStrategy.Always,
+  },
+};
+
+const dropAnimationConfig: DropAnimation = {
+  keyframes({ transform }) {
+    return [
+      { opacity: 1, transform: CSS.Transform.toString(transform.initial) },
+      {
+        opacity: 0,
+        transform: CSS.Transform.toString({
+          ...transform.final,
+          x: transform.final.x + 5,
+          y: transform.final.y + 5,
+        }),
+      },
+    ];
+  },
+  easing: 'ease-out',
+  sideEffects({ active }) {
+    active.node.animate([{ opacity: 0 }, { opacity: 1 }], {
+      duration: defaultDropAnimation.duration,
+      easing: defaultDropAnimation.easing,
+    });
+  },
+};
+
+interface Props<T> {
+  collapsible?: boolean;
+  defaultItems?: TreeItem<T>[];
+  indentationWidth?: number;
+  indicator?: boolean;
+  removable?: boolean;
+  onChange?(from: FlattenedItem<T>, to: FlattenedItem<T>): void;
+  onRemove?(item: FlattenedItem<T>): void;
+  renderItem(
+    props: Required<Omit<SortableTreeItemProps, 'renderItem'>> & {
+      item: TreeItem<T>;
+    }
+  ): React.ReactNode;
+}
+
+export function SortableTree<T = undefined>({
+  collapsible = false,
+  defaultItems = [] as TreeItem<T>[],
```

Comment:
Define this array outside as a constant.
Like this as a default arg, it could cause an infinite loop since the `[]` will not equal twice through two renders (specifically that `useEffect` using `defaultItems` in it's dependency array below).

Reply:
I think it should just be required. This was code from dnd-kit examples